README.md

```diff
@@ -2,76 +2,154 @@
+
+TiKV uses the version of the Rust toolchain specified in `RUST_VERSION`. We also make use of the `rustfmt` component.
 
 ```bash
-# Get rustup from rustup.rs, then in your `tikv` folder:
 rustup override set `tail -n 1 RUST_VERSION`
 rustup component add rustfmt-preview --toolchain `tail -n 1 RUST_VERSION`
 ```
 
-### Tikv-server software stack
-This figure represents the tikv-server software stack.
+### Building & Testing
 
-![image](images/tikv_stack.png)
+> While TiKV includes a `Makefile` with common workflows, you are also able to use `cargo` as you would a normal Rust project.
 
-- Placement driver: Placement Driver (PD) is the cluster manager of TiKV. PD periodically checks replication constraints to balance load and data automatically.
-- Store: There is a RocksDB within each Store and it stores data into local disk.
-- Region: Region is the basic unit of Key-Value data movement. Each Region is replicated to multiple Nodes. These multiple replicas form a Raft group.
-- Node: A physical node in the cluster. Within each node, there are one or more Stores. Within each Store, there are many Regions.
+At this point you should be able to build TiKV:
 
-When a node starts, the metadata of the Node, Store and Region are registered into PD. The status of each Region and Store is reported to PD regularly.
+```bash
+make build
+```
 
-### Build
+While interactively developing you may prefer using `cargo check`, which will do parse, borrow check, and lint run on your code, but not actually compile it. It is particularly handy alongside `cargo-watch` which will run a command each time you change a file.
+
+```bash
+cargo install cargo-watch
+cargo watch -s "cargo check --features dev"
+```
+
+When you're ready to test out your changes you should use the `dev` task. It will format your codebase, build with `clippy` enabled, and run tests. This should run without fail before you make a PR.
+
+```bash
+make dev
+```
+
+You can run the full test suite locally, or just run a specific test:
+
+```bash
+# Run the full suite
+make test
+# Run a specific test
+cargo test $TESTNAME
+```
+
+Any pull requests made will automatically be tested by our CI systems, so making sure the full suite passes before creating your PR is not strictly required. **All merged PRs must have passing CI tests.**
+
+### Getting the Rest of the System Working
+
+In order to get other components ([TiDB](https://github.com/pingcap/tidb) and [PD](https://github.com/pingcap/pd) working we suggest you follow the [development guide](https://github.com/pingcap/docs/blob/master/dev-guide/development.md) as you will need to (at least) have `pd-server` working alongside `tikv-server` in order to do integration level testing.
+
+## Deploying To Production
+
+**Use Ansible?** The official [Ansible playbooks](https://github.com/pingcap/tidb-ansible) are the reccomended way of deploying a cluster of TiKV nodes alongside the other components of a cluster.
```

Comment:
There is a typo `reccomended`

Reply:
Also I prefer directly providing a link to `docs/op-guide/ansible-deployment` just like the other deployment options. That document provides a link to the source repo as well if user is interested.